### PR TITLE
Vet clinic database: add "join table" for visits

### PR DESCRIPTION
**Pull request by Ibtisam34**
- [ ]  In this project, I had to add some many-to-many relationships and write more complex queries to use them to answer questions.

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -113,6 +113,69 @@ FROM species
...
 GROUP BY species.name;
 
+-- 1. Who was the last animal seen by William Tatcher?
+SELECT a.name AS animal_name
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+JOIN vets ve ON v.vet_id = ve.id
+WHERE ve.name = 'William Tatcher'
+ORDER BY v.date DESC
+LIMIT 1;
```
Comment (mikethreels) [inline queries.sql]: - [ ] the last animal seen by William Tatcher should be blossom, however, your query returns an empty row.

File: queries.sql [hunk 2]
```diff
@@ -113,6 +113,69 @@ FROM species
...
+-- 2. How many different animals did Stephanie Mendez see?
+SELECT COUNT(DISTINCT a.name) AS num_animals
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+JOIN vets ve ON v.vet_id = ve.id
+WHERE ve.name = 'Stephanie Mendez';
```
Comment (mikethreels) [inline queries.sql]: - [ ] Stephanie Mendez has seen 4 different animals according to the information provided by the assignment, your query only returns 2 please double-check.

File: queries.sql [hunk 3]
```diff
@@ -113,6 +113,69 @@ FROM species
...
+-- 3. List all vets and their specialties, including vets with no specialties.
+SELECT ve.name AS vet_name, GROUP_CONCAT(s.name SEPARATOR ', ') AS specialties
+FROM vets ve
+LEFT JOIN specializations sp ON ve.id = sp.vet_id
+LEFT JOIN species s ON sp.species_id = s.id
+GROUP BY ve.name;
```
Comment (mikethreels) [inline queries.sql]: - [ ] there is no print screen of the outcome of this query please make sure to add it as well.

File: queries.sql [hunk 4]
```diff
@@ -113,6 +113,69 @@ FROM species
...
+-- 4. List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT a.name AS animal_name
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+JOIN vets ve ON v.vet_id = ve.id
+WHERE ve.name = 'Stephanie Mendez' AND v.date BETWEEN '2020-04-01' AND '2020-08-30';
```
Comment (mikethreels) [inline queries.sql]: - [ ] this query should return Agumon and Blossom according to the provided information, please double-check.

File: queries.sql [hunk 5]
```diff
@@ -113,6 +113,69 @@ FROM species
...
+-- 5. What animal has the most visits to vets?
+SELECT a.name AS animal_name, COUNT(*) AS num_visits
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+GROUP BY a.name
+ORDER BY num_visits DESC
+LIMIT 1;
```
Comment (mikethreels) [inline queries.sql]: - [ ] according to the provided information the outcome of this query should be Boarmon, your query returns Squirtle, please check and correct.

File: queries.sql [hunk 6]
```diff
@@ -113,6 +113,69 @@ FROM species
...
+-- 6. Who was Maisy Smith's first visit?
+SELECT a.name AS animal_name
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+JOIN vets ve ON v.vet_id = ve.id
+WHERE ve.name = 'Maisy Smith'
+ORDER BY v.date
+LIMIT 1;
```
Comment (mikethreels) [inline queries.sql]: - [ ] the first visit of Maisy Smith is Boarmon according to the provided information, this query returns Charmander, please check and correct.

File: queries.sql [hunk 7]
```diff
@@ -113,6 +113,69 @@ FROM species
...
+-- 7. Details for most recent visit: animal information, vet information, and date of visit.
+SELECT a.name AS animal_name, ve.name AS vet_name, v.date
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+JOIN vets ve ON v.vet_id = ve.id
+ORDER BY v.date DESC
+LIMIT 1;
```
Comment (mikethreels) [inline queries.sql]: - [ ] it seems like the data inserted into the tables might not be entirely correct, as according to the provided data Devimon visited Stephany Mendez on 2021-05-04, your query returns Angemon visiting Maisy Smith.

File: queries.sql [hunk 8]
```diff
@@ -113,6 +113,69 @@ FROM species
...
+-- 8. How many visits were with a vet that did not specialize in that animal's species?
+SELECT COUNT(*)
+FROM visits v
+JOIN animals a ON v.animal_id = a.id
+JOIN vets ve ON v.vet_id = ve.id
+LEFT JOIN specializations sp ON ve.id = sp.vet_id AND sp.species_id = a.species_id
+WHERE sp.id IS NULL;
```
Comment (mikethreels) [inline queries.sql]: - [ ] Maisy Smith is the only vet without specialisation and she was visited 9 times, your query returns 5, please check and correct.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Riyana, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by m05e5 — APPROVED** [review]
**Hi @Ibtisam34 

You did great. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
CleverScornfulBuffalo-size_restricted

## Highlights
✔️ Descriptive PR 

## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._**

**Pull request merged**